misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/service/nodevisitors/ServiceMemberFunctionNodeVisitor.java

```diff
@@ -304,12 +319,75 @@ private List<String> getReturnTypes(FunctionDefinitionNode functionDefinitionNod
     }
 
     @Override
-    public void visit(VariableDeclarationNode variableDeclarationNode) {
+    public void visit(ObjectFieldNode objectFieldNode) {
+        Node fieldTypeName = getReferredNode(objectFieldNode.typeName());
+        if (fieldTypeName != null) {
+            Optional<Symbol> fieldTypeNameSymbol = semanticModel.symbol(fieldTypeName);
+            if (fieldTypeNameSymbol.isPresent()) {
+                ClassSymbol referredClassSymbol = getReferredClassSymbol((TypeSymbol) fieldTypeNameSymbol.get());
+                if (referredClassSymbol != null) {
+                    boolean isClientClass = referredClassSymbol.qualifiers().stream()
+                            .anyMatch(qualifier -> qualifier.equals(Qualifier.CLIENT));
+                    if (isClientClass) {
+                        String serviceId = objectFieldNode.metadata().isPresent() ?
+                                getServiceAnnotation(objectFieldNode.metadata().get().annotations(), filePath).getId() :
+                                null;
+                        Dependency dependency = new Dependency(serviceId,
+                                getClientModuleName(referredClassSymbol),
+                                getElementLocation(filePath, objectFieldNode.lineRange()));
+                        dependencyList.add(dependency);
+                    }
+                }
+            }
+        }
+    }
 
+    @Override
+    public void visit(VariableDeclarationNode variableDeclarationNode) {
+        TypedBindingPatternNode typedBindingPatternNode = variableDeclarationNode.typedBindingPattern();
+        typedBindingPatternNode.modify();
```

Comment:
Whats the purpose of using this `modify()` in here?

Reply:
@AzeemMuzammil

Reply:
This visit is redundant. I did Override to test the node and forgot to remove it. I removed it now.